.github/workflows/ci.yml

```
@@ -114,44 +135,55 @@ jobs:
       - uses: Swatinem/rust-cache@a95ba195448af2da9b00fb742d14ffaaf3c21f43
         with:
           prefix-key: "0" # change this to invalidate CI cache
+          shared-key: "cargo_nextest"
           save-if: "false" # use the cache from nextest, but don’t double-save
-      - run: cargo build -p neard --bin neard --features nightly
-      - name: run spin_up_cluster.py
-        # Note: We're not running spin_up_cluster.py for non-nightly
-        # because spinning up non-nightly clusters is already covered
-        # by other steps in the CI, e.g. upgradable.
-        run: |
-          cd pytest
-          python3 -m pip install --progress-bar off --user -r requirements.txt
-          python3 tests/sanity/spin_up_cluster.py
+      - run: pip3 install --user -r pytest/requirements.txt
+      - run: cargo build --profile quick-release -p neard --bin neard --features nightly
+      # Note: We're not running spin_up_cluster.py for non-nightly
+      # because spinning up non-nightly clusters is already covered
+      # by other steps in the CI, e.g. upgradable.
+      - run: python3 pytest/tests/sanity/spin_up_cluster.py
+        env:
+          NEAR_ROOT: "target/quick-release"
 
-      - run: cargo build --profile quick-release -p neard --bin neard
+  py_genesis_check:
+    name: "Check genesis for changes"
```

```suggestion
    name: "Genesis Changes"
```
(for uniformity, capitalize all first letters and reduce length so it hopefully fits into the github box without horizontal scrolling)